pkg/cmd/server/api/types.go

```diff
@@ -325,6 +325,9 @@ type MasterNetworkConfig struct {
 	ClusterNetworkCIDR string
 	HostSubnetLength   uint
 	ServiceNetworkCIDR string
+	// ExternalIPNetworkCIDR controls what values are acceptable for the service external IP field. If empty, no externalIP
```

Comment:
ExternalIP is a fundamental behavior of the system - by default today it is
insecure.  Config per admission control is modeled correctly for admission
controllers that are not a core part of the platform, but is not the right
model for usability or discoverability when it's a core platform feature.
The place admins will look for this control is in the MasterNetworkConfig
because it is comparable to ServiceCIDR and ClusterCIDR, defining the
acceptable range.  We will eventually move this into a flag on apiserver
and do the same validation, when we do, the config should come from our
core object, not an admission controller.

ExternalIP is a security risk, and we cannot ship with it enabled by
default, therefore we must disable by default, and existing users will be
broken.  We cannot allow through inaction for users to have a vulnerability
here.  Release notes will be updated
https://github.com/openshift/openshift-docs/issues/1700

On Mon, Mar 7, 2016 at 6:53 AM, David Eads <notifications@github.com> wrote:

> In pkg/cmd/server/api/types.go
> <https://github.com/openshift/origin/pull/7810#discussion_r55198749>:
>
> > @@ -325,6 +325,9 @@ type MasterNetworkConfig struct {
> >  	ClusterNetworkCIDR string
> >  	HostSubnetLength   uint
> >  	ServiceNetworkCIDR string
> > +	// ExternalIPNetworkCIDR controls what values are acceptable for the service external IP field. If empty, no externalIP
>
> That's a strange way to deal with empty string and it makes it impossible
> to default to existing behavior, doesn't it?
>
> If you moved this to the admission config section, then no config means
> "allow all" (existing behavior) and config with empty string can mean
> "restrict all". That seems to be more intuitive.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/7810/files#r55198749>.
>